### ISSUE TITLE
YoloV5 train seems to ignore half the validation set?

### ISSUE THREAD

**Issue by duckduck-sys** [ultralytics/yolov5#5035]
My train.yaml file:

```
train: /data/train_set.txt
val: /data/val_set.txt
nc: 3
names: ['Athlete', 'Referee', 'Time']
```

For which i run:

`python train.py --img 640 --batch 32 --epochs 200 --data ./train.yaml --weights yolov5s6.pt --device 0 --hyp hyp.scratch-p6.yaml`

It finds
```

train: Scanning '/data/train_set.cache' images and labels... 68991 found, 0 missing, 1914 empty, 0 corrupted
val: Scanning '/data/val_set.cache' images and labels... 17253 found, 0 missing, 508 empty, 0 corrupted
```

Which is all correct, everything seems to be working fine and the output model after training is very good. But, during each epoch it does 2156 steps for training, and only 270 steps for validation

Training : 32 images per batch * 2156 batch steps per epoch = 68992 images per epoch
Validation : 32 images per batch * 270 batch steps per epoch = 8640 images per epoch

I.e. only half the provided validation images seem to be used. Am i missing something? It appears that no matter how many or how few validation images i provide, only half of them are used.

**Comment by glenn-jocher** [comment]
@duckduck-sys everything works correctly. All validation images are used of course.
https://github.com/ultralytics/yolov5/blob/b0ade48457fb8b1fc1acd9e1a34bb50e6b7205da/train.py#L353-L363

**Comment by duckduck-sys** [comment]
Ah yes, of course. Thanks!
